core/src/main/java/org/apache/calcite/sql2rel/RelDecorrelator.java

```diff
@@ -224,13 +241,9 @@ private void setCurrent(RelNode root, LogicalCorrelate corRel) {
     }
   }
 
-  private RelBuilderFactory relBuilderFactory() {
-    return RelBuilder.proto(relBuilder);
```

Comment:
@julianhyde , @vvysotskyi , just in case, this line did not work as expected.
In fact, `RelBuilder.proto(...)` should receive `RelFactory` kind of arguments, not a ready `RelBuilder`.
In other words, this `relBuilderFactory`  method always produced logical RelBuilder since by default it uses LogicalFactories.

Reply:
The intention of this method was to avoid passing `RelBuilderFactory` to the `decorrelateQuery()` method since `relBuilder` is always created using this factory.